src/util.ts

```diff
@@ -41,6 +42,38 @@ interface PackageJson {
   version: string;
 }
 
+export interface ClassOf<T, Config> {
+  new(logger: Logger, config: Config): T;
+  prototype: T;
+  name: string;
+}
+
+/**
+ * An class that provides access to a singleton.
+ * Instances of this type should only be constructed in module scope.
+ */
+export class Singleton<T, Config> {
+  private singleton: T|null = null;
```

Comment:
suggestion: use a Symbol for better privacy?

Reply:
Done